### DISCUSSION TITLE
How to switch from Dolby Vision profile 8 to 5

### DISCUSSION THREAD

**Discussion by SkinnyPeteBB** [quietvoid/dovi_tool#210] [Q&A]
I have this file that has: **HDR format : Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible**. I just want to have a file that can be played properly on my TV. A file with: **HDR: Dolby Vision, version 1.0, dvhe.05.06, Bl+RPU** is reproduced correctly, so I think the first is DV profile 8, the second profile 5. How do I go from 8 to 5 in mp4?  Maybe I am doing it wrong and they are not the right profiles, you tell me what I should do.

**Comment by quietvoid** [comment]
It's not possible and requires transcoding.

**Comment by SkinnyPeteBB** [comment]
> It's not possible and requires transcoding.

Is there a way to get it to play correctly on the TV?  The file is from a hulu series so there's some way to play it right?

**Reply by quietvoid** [reply]
Can you be more specific as to how it doesn't play on your TV? Also please clarify your TV model.

**Reply by hyllm** [reply]
try run HEVC stream through "dovi_tool convert --mode 0 --drop-hdr10plus --input input.hevc --output output.hevc" (or something like that... "--drop-hdr10plus" is important here)  and the mux to mkv/mp4. i saw TV's which weren't playing Dolby Vision due to HDR10+ metadata (i.e. Philips 65OLED806)

**Reply by SkinnyPeteBB** [reply]
> Can you be more specific as to how it doesn't play on your TV? Also please clarify your TV model.

I have a tv sony bravia xf90. If I try to play the file on the system media player it plays in hdr10. If I try to play it on Nova Video player I see the DV logo at the top right, but it is reproduced with the wrong colors or black and white.

**Reply by SkinnyPeteBB** [reply]
> try run HEVC stream through "dovi_tool convert --mode 0 --drop-hdr10plus --input input.hevc --output output.hevc" (or something like that... "--drop-hdr10plus" is important here)  and the mux to mkv/mp4. i saw TV's which weren't playing Dolby Vision due to HDR10+ metadata (i.e. Philips 65OLED806)

Please could you explain me better what to do.

**Reply by quietvoid** [reply]
I'm not sure if it's caused by  HDR10+ but the black and white problem is a known issue for Sony TVs.
Anyways, you can't play Dolby Vision in MKV on the  regular player as it only supports MP4.

**Reply by SkinnyPeteBB** [reply]
> I'm not sure if it's caused by  HDR10+ but the black and white problem is a known issue for Sony TVs.
> Anyways, you can't play Dolby Vision in MKV on the  regular player as it only supports MP4.

Yes in fact I changed container to MP4 with ffmpeg, but nothing. Maybe I did something wrong in converting it from MKV to MP4.

**Reply by quietvoid** [reply]
Hmm well if you checked the MP4 mediainfo and it had the Dolby Vision info, it should have worked.

FFmpeg requires a specific flag, you can also try MP4Box from GPAC.

**Reply by SkinnyPeteBB** [reply]
I don't know what to do, I think I'll just look at it in hdr10. Thanks for your help. I found this gui for dovi_tool, but I would not know how to use it for my problem.![IMG_20230211_180501.jpg](https://user-images.githubusercontent.com/115431386/218271040-ec8a321c-59f5-429f-9346-bdea579ba7a2.jpg)

**Reply by SkinnyPeteBB** [reply]
> Hmm well if you checked the MP4 mediainfo and it had the Dolby Vision info, it should have worked.
> 
> FFmpeg requires a specific flag, you can also try MP4Box from GPAC.

I tried mp4box and now I have the file with DV profile 5 and also the Sony system player recognizes it as such. But it is always reproduced wrong.

**Reply by quietvoid** [reply]
Not much we can do about it here, complain to Sony.

**Reply by SkinnyPeteBB** [reply]
> Not much we can do about it here, complain to Sony.

Okay, thanks.

**Reply by NeedsMoar** [reply]
> > Hmm well if you checked the MP4 mediainfo and it had the Dolby Vision info, it should have worked.
> > FFmpeg requires a specific flag, you can also try MP4Box from GPAC.
> 
> I tried mp4box and now I have the file with DV profile 5 and also the Sony system player recognizes it as such. But it is always reproduced wrong.

That's because you don't _want_ the file in profile 5, because it isn't.  You can feed mp4muxer any random profile you want but if it isn't correct you'll get garbage...  (When apple first introduced Profile 8.4, I forced an 8.1 file to think it was 8.4 using mp4muxer.  The colors were godawful horrors because the file wasn't really an HLG but the LG recognized it as dolby vision and played it which was all I was testing since I don't own any apple devices).

>HDR format : Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible

The first thing I'd try here is stripping that HDR10+ data off in case the TV is either preferring it or getting confused.  I think Sony actually supports HDR10+ (?) so if you're trying to get the dolby vision to show up getting rid of that at least narrows something down.  

That, and try remuxing to an .m2ts.   Someone else was having issues with .mp4 on Sony's player.  I don't know if .m2ts will help but it's pretty baseline and tsmuxer doesn't screw it up.

**Reply by SkinnyPeteBB** [reply]
> > > Hmm well if you checked the MP4 mediainfo and it had the Dolby Vision info, it should have worked.
> > > FFmpeg requires a specific flag, you can also try MP4Box from GPAC.
> > 
> > I tried mp4box and now I have the file with DV profile 5 and also the Sony system player recognizes it as such. But it is always reproduced wrong.
> 
> That's because you don't _want_ the file in profile 5, because it isn't.  You can feed mp4muxer any random profile you want but if it isn't correct you'll get garbage...  (When apple first introduced Profile 8.4, I forced an 8.1 file to think it was 8.4 using mp4muxer.  The colors were godawful horrors because the file wasn't really an HLG but the LG recognized it as dolby vision and played it which was all I was testing since I don't own any apple devices).
> 
> >HDR format : Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible
> 
> The first thing I'd try here is stripping that HDR10+ data off in case the TV is either preferring it or getting confused.  I think Sony actually supports HDR10+ (?) so if you're trying to get the dolby vision to show up getting rid of that at least narrows something down.  
> 
> That, and try remuxing to an .m2ts.   Someone else was having issues with .mp4 on Sony's player.  I don't know if .m2ts will help but it's pretty baseline and tsmuxer doesn't screw it up.
> 

However the TV doesn't support dolby vision profile 8 or even hdr10+, which is only supported by a few TVs such as samsung TVs.  I can try to remove the hdr10+ metadata, how can I do that? I can also try converting them to .m2ts or .ts with tsmuxer.

**Comment by Dunge** [comment]
I too have the same issue trying to play new released movies on my LG C8 (2018 model) through Plex. Seems to support Profile 5 perfectly, but play as normal HDR with Profile 8 files.

Would love a way to convert it.

**Reply by quietvoid** [reply]
> I too have the same issue trying to play new released movies on my LG C8 (2018 model) through Plex. Seems to support Profile 5 perfectly, but play as normal HDR with Profile 8 files.

The TV supports both profiles, it's either a Plex issue or your files are MKV.

**Reply by Dunge** [reply]
Where do you see that? Every specs I've seen only say "Dolby Vision: yes".

In any case, I can confirm that it in fact do not. I always grab mp4 when available, otherwise convert using ffmpeg, but the result is always the same. When the file is described as dvhe.07 or dvhe.08, it never trigger DV and use the normal HDR fallback. I repeated the experience with dozens of different sources and it's always the case, but more and more movies are just release in 7/8 now. Hell, I'm currently watching one. When it's described as dhve.05, it trigger the dolby vision properly (like it was for tons of movies I watched over the years before that).

And I don't think it's a Plex issue. I set up a DLNA server and used the official LG "Photo & Video" app and it does the same.

Maybe it's when it's a dual layer (with HDR fallback) the TV is too stupid and decide to use that instead, but I never saw Profile 7/8 content without that dual layer thing to try.

**Reply by quietvoid** [reply]
> In any case, I can confirm that it in fact do not.

I've owned a C8 and it definitely did.
It just has particuliar requirements for which MP4 muxer is supported.
Only Dolby's old mp4muxer worked, and it even supports profile 7 with the internal player.

Anyways, you can probably google info and tips about it. Good luck.
